packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -808,12 +808,17 @@ export class FocusZone extends BaseComponent<IFocusZoneProps, {}> implements IFo
       // We shouldn't lose focus in the following cases:
       // 1. There is range selected.
       // 2. When selection start is larger than 0 and it is backward.
-      // 3. when selection start is not the end of lenght and it is forward.
+      // 3. when selection start is not the end of length and it is forward.
       // 4. We press any of the arrow keys when our handleTabKey isn't none or undefined (only losing focus if we hit tab)
-      if (isRangeSelected ||
+      // and if shouldInputLoseFocusOnArrowKey is defined, if scenario prefers to not loose the focus which is determined by calling the
+      // callback shouldInputLoseFocusOnArrowKey
+      if (
+        isRangeSelected ||
         (selectionStart > 0 && !isForward) ||
         (selectionStart !== inputValue.length && isForward) ||
-        !!this.props.handleTabKey) {
+        (!!this.props.handleTabKey && !(this.props.shouldInputLoseFocusOnArrowKey
+          && this.props.shouldInputLoseFocusOnArrowKey(element)))
```

Comment:
is it possible to add a unit test to cover this scenario so we don't break it in the future?

Reply:
(it's super easy, go to the packages/oufr folder, run `npm run start-test`, edit FocusZone.test.tsx and add a test.

Reply:
I get below error when I run start-test 

![image](https://user-images.githubusercontent.com/22942899/37534050-e2394d1a-2900-11e8-966b-cab55561ec29.png)


Also what do you mean by packages/oufr folder

Reply:
Nevermind Joe told me what oufr means :D

Reply:
Added test